dbms/src/Storages/KVStore/MultiRaft/Spill/LargeTxnDefaultCf.h

```diff
@@ -0,0 +1,124 @@
+        return Map::value_type(
+            std::move(tidb_pk),
+            Value(
+                std::make_shared<const TiKVKey>(std::move(key)),
+                std::make_shared<const TiKVValue>(std::move(value))));
+    }
+
+    static std::shared_ptr<const TiKVValue> getTiKVValue(const Map::const_iterator & it)
+    {
+        return std::get<1>(it->second);
+    }
+};
+
+struct LargeTxnDefaultCf
+{
+    using Trait = LargeDefaultCFDataTrait;
+    using Inner = RegionCFDataBase<LargeDefaultCFDataTrait>;
+    using Level1Key = Timestamp;
+    using Level2Key = typename Trait::Key; // Actually RawTiDBPK
+    using Value = typename Trait::Value;
+    using Map = typename std::unordered_map<Level1Key, std::shared_ptr<Inner>>;
+    using InnerMap = Inner::Map;
+    using Data = Map;
+    using Pair = std::pair<Level2Key, Value>;
+    using Status = bool;
+
+    using ConstTwoLevelIt = std::optional<std::tuple<Map::const_iterator, InnerMap::const_iterator>>;
+    using TwoLevelIt = std::optional<std::tuple<Map::iterator, InnerMap::iterator>>;
+
+    ConstTwoLevelIt find(const Level2Key & key, const Level1Key & ts) const;
+
+    static std::shared_ptr<Inner> & mustGet(LargeTxnDefaultCf & cf, const Level1Key & key);
+
+    RegionDataRes insertWithTs(TiKVKey && key, TiKVValue && value, Timestamp ts, DupCheck mode = DupCheck::Deny);
+    RegionDataRes insert(TiKVKey && key, TiKVValue && value, DupCheck mode = DupCheck::Deny);
+
+    static size_t calcTiKVKeyValueSize(const Inner::Value & value);
+    static size_t calcTiKVKeyValueSize(const TiKVKey & key, const TiKVValue & value);
+
+    // Serves "Del" raft command.
+    size_t remove(const Level2Key & key, const Level1Key & ts, bool quiet = false);
+    // Just remove the given key, if any.
+    void erase(const Level2Key & key, const Level1Key & ts);
+    void erase(const ConstTwoLevelIt & it);
+    static bool cmp(const Map & a, const Map & b);
+
+    bool operator==(const LargeTxnDefaultCf & cf) const;
+
+    size_t getSize() const;
+
+    size_t getTxnCount() const;
+
+    LargeTxnDefaultCf() = default;
+    LargeTxnDefaultCf(LargeTxnDefaultCf && region)
+        : txns(std::move(region.txns))
+    {}
+
+    LargeTxnDefaultCf & operator=(LargeTxnDefaultCf && region)
+    {
+        txns = std::move(region.txns);
+        return *this;
+    }
+
+    size_t splitInto(const RegionRange & range, LargeTxnDefaultCf & new_region_data);
+    size_t mergeFrom(const LargeTxnDefaultCf & ori_region_data);
+
+    size_t serializeMeta(WriteBuffer & buf) const;
+    size_t serialize(WriteBuffer & buf) const;
+    static size_t deserializeMeta(ReadBuffer & buf);
+    static size_t deserialize(ReadBuffer & buf, size_t txn_count, LargeTxnDefaultCf & new_region_data);
+
+    const Data & getData() const { return txns; }
+    Data & getDataMut() { return txns; }
+    const Inner & getTxn(const Level1Key & ts) const { return *txns.at(ts); }
+    Inner & getTxnMut(const Level1Key & ts) { return *txns.at(ts); }
```

Comment:
Seems not used. And what if `ts` is not exist?

Reply:
Throws a standard error is ok. I am just curious where is be used.

Reply:
changed into DB::Exception, since std exception is not as respected as DB exception.